app/Repository/Eloquent/AuthRepository.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Repository\Eloquent;
+
+use App\Models\User;
+use App\Repository\BaseRepository;
+
+class AuthRepository extends BaseRepository{
+
+    protected $user;
+
+    public function __construct(User $user)
+    {
+        $this->user = $user;
+    }
+
+    public function signUp($request){
+        $request['password'] = bcrypt($request['password']);
+
+        if(!$user = $this->user->create($request)){
+            return [
+                "status" => $this->isUnsuccessful(),
+                "message" => "SignUp Failed"
+            ];
+        }
+
+        return [
+            "status" => $this->isSuccessful(),
+            "user" => $user,
+        ];
+    }
+
+    public function signIn($request){
```

Comment:
for here, the codes here are too much, ensure that $request which should be named $data has only email and password in the array so that it becomes: 
`auth()->attempt()->($data)`

so everything becomes:

```
 if(! auth()->attempt($data){ return false}
return auth()->login($this->user->where("email", $request['email'])->first());
```